lsr-role2collection.py

```diff
@@ -0,0 +1,418 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# (c) 2020 Matt Martz <matt@sivel.net>
+# GNU General Public License v3.0+
+#     (see https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# Usage:
+# lsr-role2collection.py [--namespace NAMESPACE]
+#                        [--collection COLLECTION]
+#                        --src-path SRC_PATH
+#                        --dest-path DEST_PATH
+#                        --role ROLE
+#                        [-h]
+
+import argparse
+import os
+import re
+import fnmatch
+from shutil import copytree, copy2, ignore_patterns
+
+from pathlib import Path
+
+ROLE_DIRS = (
+    'defaults',
+    'files',
+    'handlers',
+    'meta',
+    'tasks',
+    'templates',
+    'vars',
+)
+
+PLUGINS = (
+    'action_plugins',
+    'become_plugins',
+    'cache_plugins',
+    'callback_plugins',
+    'cliconf_plugins',
+    'connection_plugins',
+    'doc_fragments',
+    'filter_plugins',
+    'httpapi_plugins',
+    'inventory_plugins',
+    'library',
+    'lookup_plugins',
+    'module_utils',
+    'netconf_plugins',
+    'shell_plugins',
+    'strategy_plugins',
+    'terminal_plugins',
+    'test_plugins',
+    'vars_plugins'
+)
+
+TESTS = (
+    'tests',
+    'molecule',
+    '.travis',
+)
+
+DOCS = (
+    'docs',
+    'design_docs',
+)
+
+ALL_DIRS = ROLE_DIRS + PLUGINS + TESTS + DOCS
+
+IMPORT_RE = re.compile(
+    br'(\bimport) (ansible\.module_utils\.)(\S+)(.*)$',
+    flags=re.M
+)
+FROM_RE = re.compile(
+    br'(\bfrom) (ansible\.module_utils\.?)(\S+)? import (\(*(?:\n|\r\n)?)(.+)$',
+    flags=re.M
+)
+FROM2DOTS_RE = re.compile(
+    br'(\bfrom) \.\.(module_utils\.)(\S+) import (.+)$',
+    flags=re.M
+)
+
+
+def dir_to_plugin(v):
+    if v[-8:] == '_plugins':
+        return v[:-8]
+    elif v == 'library':
+        return 'modules'
+    return v
+
+
+# python lsr-role2collection.py /src_path/linux-system-roles/logging /dest_path/ansible_collections/redhat/system_roles
+# positional arguments:
+#  ROLE_PATH        Path to a role to migrate
+#  COLLECTION_PATH  Path to collection where role should be migrated
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    '--namespace',
+    type=str,
+    default=os.environ.get("COLLECTION_NAMESPACE", "redhat"),
+    help='Collection namespace; default to redhat',
+)
```

Comment:
@pcahyna 
> how will this work in the Collection world? Will there be a single namespace for the whole collection? Can we do this in a way that it prepares for future work on collections? @nhosoi

Not sure this answers your question, but when converting a role to a collection with the tool `lsr-role2collection.py`, you could set an option `--namespace YOURNAMESPACE`. If you don't give the option, default to `redhat` (for now, we could change it to something else, e.g., `fedora`).